cmd/car/extract.go

```diff
@@ -0,0 +1,192 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path"
+
+	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-unixfsnode"
+	"github.com/ipfs/go-unixfsnode/data"
+	"github.com/ipfs/go-unixfsnode/file"
+	"github.com/ipld/go-car/v2/blockstore"
+	dagpb "github.com/ipld/go-codec-dagpb"
+	"github.com/ipld/go-ipld-prime"
+	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
+	basicnode "github.com/ipld/go-ipld-prime/node/basic"
+	"github.com/urfave/cli/v2"
+)
+
+// ExtractCar pulls files and directories out of a car
+func ExtractCar(c *cli.Context) error {
+	if !c.IsSet("file") {
+		return fmt.Errorf("a file source must be specified")
+	}
+	outputDir, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+	if c.Args().Len() > 0 {
+		outputDir = c.Args().First()
+	}
+
+	bs, err := blockstore.OpenReadOnly(c.String("file"))
```

Comment:
Would it make sense to [clean](https://pkg.go.dev/path/filepath#Clean) the path before using?

Reply:
it should be fine to open and let the OS clean implicitly, right?